evm_loader/program/src/solidity_account.rs

```diff
@@ -44,11 +44,17 @@ impl<'a> SolidityAccount<'a> {
                 return f(&data[offset..offset+code_size as usize])
             }
         }*/
-        if self.code_data.is_some() && self.code_data.as_ref().unwrap().0 > 0 {
-            let some_data = self.code_data.as_ref().unwrap().1.clone();
-            let data = some_data.borrow();
-            let code_size = self.code_data.as_ref().unwrap().0 as usize;
-            f(&data[Contract::SIZE..Contract::SIZE+code_size])
+        if self.code_data.is_none() {
+            return f(&[])
+        }
+
+        let contract_data = &self.code_data.as_ref().unwrap().0;
+        let code_size = contract_data.code_size as usize;
+        let contract_data = AccountData::Contract(contract_data.clone());
```

Comment:
I don't like need of copying in situations like this.
But if i use AccountData::Contract instead of Contract appears a lot of match AccountData to Contract  constructions.